cluster/gce/manifests/etcd.manifest

```diff
@@ -64,13 +64,17 @@
       }
     ],
     "livenessProbe": {
-      "httpGet": {
-        "host": "127.0.0.1",
-        "port": {{ etcd_livenessprobe_port }},
-        "path": "/health"
+      "exec": {
+        "command": [
+          "/bin/sh",
+          "-c",
+          "exec /usr/local/bin/etcdctl --endpoints=127.0.0.1:{{ port }} {{ etcdctl_certs }} --command-timeout=15s endpoint health"
+        ]
       },
       "initialDelaySeconds": {{ liveness_probe_initial_delay }},
-      "timeoutSeconds": 15
+      "timeoutSeconds": 15,
+      "periodSeconds": 15,
```

Comment:
I'm not sure if I understand the motivation to increase the periodSeconds (10s -> 15s). I'd propose the opposite, let's lower it to accommodate for the increased failureThreshold (3->5). Would you mind explaining your reasoning?

Reply:
I'm trying to avoid overlapping probe attempts, i.e. not start the next probe if the previous one hasn't finished. While I increase timeout, I need to increase probe interval as well.

Separate, independent probe attempts provide us a better signal than overlapping ones.

Reply:
If the probe is as expensive as individual read, I think we could afford overlapping ones.

Reply:
I think it doesn't work this way. It's not possible to have overlapping probes, see the code - https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/kubelet/prober/worker.go#L151-L160

Reply:
As discussed offline, I changed this to 5s.

For posterity, the reason for that based on our experiments etcdctl endpoint health blocks for timeoutSecond in most of the unhealthy scenarios and that case it will require timeoutSecond * failureThreshold of etcd unhealthiness to trigger restart. The advantage of setting smaller value for probeSecond is that this reduces average time to detect first unhealthy probe (e.g. with probeSecond=15, if we finished some probe at time 0, and etcd becomes unhealthy at time 1, we will need to wait next 14 seconds before we start probing it, while with probeSecond we will start probing at time t=5).

Reply:
SGTM